DSpace/dspace-angular
# Fixed `ngOnChanges` not working for themed components & decorator components + minor fixes

## References
* Fixes #2201

## Description
I looked over the code again and I saw that this issue was already partially fixed in PR https://github.com/DSpace/dspace-angular/pull/1512. The `ngOnChanges()` from the child component is already being called in the `ThemedComponent` but in some rare cases the `ngOnChanges()` is triggered before the themed child component is created by the themed wrapper. When the `ngOnChanges()` isn't called multiple times this can lead to the child component's `ngOnChanges()` not being triggered at all. To fix this the component will be created in the `ngOnChanges()` if it doesn't exist yet and the child's `ngOnChanges` will be manually triggered afterwards as well.

This is also the case for the components who are rendered using the custom decorators, the rendered component's `ngOnChanges` isn't called their either when changes are detected in the loader component.

## Instructions for Reviewers
List of changes in this PR:
- Made `ThumbnailComponent` themeable
- Modified `ThemedComponent.ngOnChanges()` to initiated the themed child component when it doesn't exists yet, because previously when it didn't exist it would just skip it which caused `ngOnChanges` sometimes from not being called.
- Fixed an issue where the `src` would be `undefined` when the thumbnail isn't defined (this throws a 404 error in the network tab because it tries to retrieve the url `http://localhost:4000/undefined`, this can be seen on demo if you go to the full item page of an item)
- Added missing `offset-3` class to `CommunitySearchResultListElementComponent` to fix listable communities in the search tab that aren't aligned on mobile search page for listable communities.
- Made changes to `ListableObjectComponentLoaderComponent`, `MetadataRepresentationLoaderComponent`, `ClaimedTaskActionsLoaderComponent` to fix the same problem. The `ngOnChanges` from the components with those decorators would not automatically be called anymore.

**Guidance for how to test this PR:**
`ThumbnailComponent`
* Set the theme to `custom` theme in `config.yml`
* Uncomment the `CustomEagerThemeModule` in `src/themes/eager-themes.module.ts`
* Test if the `ThumbnailComponent` in the `custom` theme folder still works like before (on item pages and on the listable search components)
* Test by adding a debug point in the `ngOnChanges()` method or adding a `console.log()` and checking that it passed through the `ngOnChanges()` method

The custom components loaders:
* Find a component that is rendered by `ListableObjectComponentLoaderComponent`, `MetadataRepresentationLoaderComponent` and `ClaimedTaskActionsLoaderComponent` and add in each of those components a new `ngOnChanges` and check that they are called.
* Check that all the other components using this decorator still work as before

## Checklist
